vm/src/compiler.rs

```diff
@@ -33,6 +33,8 @@ enum FieldAccess {
 #[derive(Debug)]
 pub struct CompiledFunction {
     pub args: VmIndex,
+    /// The maximum possible number of stack slots needed for this function
+    pub required_stack_size: VmIndex,
```

Comment:
`max_stack_size` may be clearer, going to replace uses of `stack_limit` since it makes no indication towards what limit it is.